### PR TITLE
Add magics credentials monkeypatching if the kernel has bigquery inte…

### PR DESCRIPTION
**Pull request by vimota**
…gration.

### CODE REVIEW

File: patches/sitecustomize.py [hunk 1]
```diff
@@ -3,41 +3,50 @@
...
+                return bq_client(*args, **kwargs)
+
+        # Monkey patches BigQuery client creation to use proxy or user-connected GCP account.
+        # Deprecated in favor of Kaggle.DataProxyClient().
+        # TODO: Remove this once uses have migrated to that new interface.
+        bq_client = bigquery.Client
+        bigquery.Client = lambda *args, **kwargs:  monkeypatch_bq(
+            bq_client, *args, **kwargs)
+
+init()
```
Comment (rosbo) [inline patches/sitecustomize.py]: Don't you want to call init only if the kernel has the integration?

Comment (vimota) [reply patches/sitecustomize.py]: My idea here was "init the sitecustomize logic", basically leave this program unchanged in logic but just make it callable to reproduce starting up in tests. I wanted to be able to unit test if "KAGGLE_KERNEL_INTEGRATIONS" is set for a Kernel will the the monkeypatching logic behave correctly on startup.

Comment (vimota) [reply patches/sitecustomize.py]: So, if I were to only call init if the kernel has the integration, then I wouldn't be to test that logic in the unit test.

Comment (rosbo) [reply patches/sitecustomize.py]: Makes sense. Thanks for the clarification.

**Pull request merged**